File: src/main/java/me/centralhardware/znatoki/telegram/statistic/validate/FioValidator.java
package me.centralhardware.znatoki.telegram.statistic.validate;

import io.vavr.control.Either;
import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Client;
import me.centralhardware.znatoki.telegram.statistic.service.ClientService;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class FioValidator implements Validator<String, String> {
    private final ClientService clientService;

    @Override
    public Either<String, String> validate(String fio) {
        Client exists = clientService.findByFioAndId(fio);
        if (exists == null) {
            return Either.left("ФИО не найдено");
        }
        return Either.right(fio);
    }
}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/validate/FioValidator.kt
package me.centralhardware.znatoki.telegram.statistic.validate

import arrow.core.Either
import me.centralhardware.znatoki.telegram.statistic.mapper.ClientMapper
import org.springframework.stereotype.Component

@Component
class FioValidator(private val clientMapper: ClientMapper) : Validator<String, String> {
    override fun validate(value: String): Either<String, String> = if (clientMapper.existsByFio(value)){
        Either.Right(value)
    } else{
        Either.Left("ФИО не найдено")
    }
}